demo/lightning/text-based/distributed/model.py

```diff
@@ -206,15 +221,20 @@ def __iter__(self):
         end_point = start_point + files_per_node
         if worker_info is None:
             print("Single-process data loading detected", flush=True)
-            for bytes_content in df_iter.dataflux_core.download.dataflux_download_lazy(
+            if self.no_dataflux:
```

Comment:
The `__iter__` method has quite a bit of conditional branching logic. It's a relatively simple class, so I would suggest considering separating this into two separate classes (e.g. `ParquetDatafluxIterableDataset` and `ParquetFsspecIterableDataset`) and using the flag to decide which dataset class to construct in the main method. Same suggestion for the single node demo too.

Reply:
We do have tasks slated to make this a standalone dataset in the future, which will probably incorporate the changes you're requesting here.